wallet/src/de/schildbach/wallet/ui/dashpay/PlatformRepo.kt

```diff
@@ -576,8 +583,20 @@ class PlatformRepo private constructor(val walletApplication: WalletApplication)
 
     fun initBlockchainIdentity(blockchainIdentityData: BlockchainIdentityData, wallet: Wallet): BlockchainIdentity {
         val creditFundingTransaction = blockchainIdentityData.findCreditFundingTransaction(wallet)
-        if (creditFundingTransaction != null) {
-            return BlockchainIdentity(platform, creditFundingTransaction, wallet).apply {
+        val blockchainIdentity = if (creditFundingTransaction != null) {
+            BlockchainIdentity(platform, creditFundingTransaction, wallet)
+        } else {
+            val blockchainIdentity = BlockchainIdentity(platform, 0, wallet)
+            if (blockchainIdentityData.creationState >= BlockchainIdentityData.CreationState.DONE) {
+                blockchainIdentity.apply {
+                    uniqueId = Sha256Hash.wrap(Base58.decode(blockchainIdentityData.userId))
+                }
+            } else {
+                return blockchainIdentity
```

Comment:
we return the blockchainIdentity because this is during the registration process and there is no other data to load.